### DISCUSSION TITLE
error to use bbram on stm32

### DISCUSSION THREAD

**Discussion by lmikl** [zephyrproject-rtos/zephyr#55486] [Q&A] [answered]
HI,

i would like used the bbram on a stm32f4 but i have a time an compilation error

I have configured an overlay for the board nucleo-f429zi :
 / {
	aliases {
		backup-register = &bbram;
	};
};

&bbram {
	status = "okay";
};

write this onfig in the prj.comf : CONFIG_BBRAM=y

and i have wrote this code to use it : 

static const struct device *backup_reg = NULL;

int main(void)
{
     backup_reg = DEVICE_DT_GET(DT_ALIAS(backup_register));

    if(backup_reg == NULL){
        LOG_INF("backup_reg device was not found");
    }
    else
        LOG_INF("backup_reg found");

    if (!device_is_ready(backup_reg)) {
		LOG_INF("backup_reg not ready");
    }
}

But i have all time this error : undefined reference to `__device_dts_ord_105'.

If someone can help me.
thanks in advance.

**Comment by erwango** [comment, marked as answer]
Since BBRAM is in RTC, you might need to enable RTC as well: `CONFIG_COUNTER=y`
with &rtc{status = "okay"};

**Reply by lmikl** [reply]
really thanks. it works.

**Reply by erwango** [reply]
Great. We'll need to work on this to enhance provide a meaningful error message.
